collectors/jiraffe/convertors.py

```diff
@@ -0,0 +1,180 @@
+"""
+transform Jira issue into OSIDB tracker model
+"""
+import uuid
+from functools import cached_property
+
+from django.conf import settings
+
+from osidb.core import generate_acls, set_user_acls
+from osidb.models import Tracker
+
+from ..utils import (
+    tracker_parse_update_stream_component,
+    tracker_summary2module_component,
+)
+
+
+class TrackerConvertor:
+    """
+    generic raw tracker to OSIDB tracker convertor
+
+    this class transforms raw data from a unified raw format into
+    proper Tracker model records and saves them into the database
+    """
+
+    def __init__(
+        self,
+        tracker_data,
+    ):
+        self._raw = tracker_data
+        # important that this is last as it might require other fields on self
+        self.tracker_data = self._normalize()
+        # set osidb.acl to be able to CRUD database properly and essentially bypass ACLs as
+        # celery workers should be able to read/write any information in order to fulfill their jobs
+        set_user_acls(
+            settings.PUBLIC_READ_GROUPS
+            + [
+                settings.PUBLIC_WRITE_GROUP,
+                settings.EMBARGO_READ_GROUP,
+                settings.EMBARGO_WRITE_GROUP,
+            ]
+        )
+
+    @property
+    def type(self):
+        """
+        concrete tracker type has to be specified in the child classes
+        """
+        raise NotImplementedError
+
+    @property
+    def _normalize(self):
+        """
+        to be implemented in the child classes
+        """
+        raise NotImplementedError
+
+    def _gen_tracker_object(self, affect) -> Tracker:
+        """
+        generate Tracker object from raw tracker data
+        """
+        # there maybe already existing tracker from the previous sync
+        # if this is the periodic update however also when the flaw bug
+        # has multiple CVEs the resulting flaws will share the trackers
+        tracker = Tracker.objects.create_tracker(
+            affect=affect,
+            _type=self.type,
+            external_system_id=self.tracker_data["external_system_id"],
+            status=self.tracker_data["status"],
+            resolution=self.tracker_data["resolution"],
+            ps_update_stream=self.tracker_data["ps_update_stream"],
+            meta_attr=self.tracker_data,
+            acl_read=self.acl_read,
+            acl_write=self.acl_write,
+        )
+        # eventual save inside create_tracker would
+        # override the timestamps so we have to set them here
+        tracker.created_dt = self.tracker_data["created_dt"]
+        tracker.updated_dt = (
+            self.tracker_data["updated_dt"] or self.tracker_data["created_dt"]
+        )
+        return tracker
+
+    def convert(self, affect=None) -> Tracker:
+        return self._gen_tracker_object(affect)
+
+
+class TrackerIssueConvertor(TrackerConvertor):
```

Comment:
I'd just call them `BugzillaTrackerConvertor` and `JiraTrackerConvertor`, because that's what you want to convey but make it less understandable by using `Issue` and `Bug`.

Reply:
I will rename.